Extensions/separator.js

```
@@ -1,212 +1,570 @@
+// # Motivation
+// Some of us like to catch up with all the posts that we’ve missed since we last checked our dashboard.
+// Placing a visual separator to show where we left off helps with that.
+//
+// # Approach
+// For each context (dashboard, specific blog, or specific tag) we track the newest post we’ve previously witnessed.
+// This is our “goal post” (above which we’ll place the separator).
+// When we return to the context, we track the set of posts that we’ve witnessed that’s newer than the goal post.
+// Once we reach the goal post, we place the separator.
+// We then reset the goal post (to the newest post in the contiguous interval of witnessed posts that’s touching the goal post).
+//
+// # Implementation-specific Caveats
+//   - We don’t actually track the set of discrete witnessed posts; we track a set of disjoint intervals representing contiguously witnessed posts.
+//     This approach should tolerate deleted posts.
+//     However, a witnessed post that’s subsequently been edited will still be treated as witnessed.
+//   - When endless scrolling is off, merging of seemingly disjoint intervals from adjacent pages relies on URL hints.
+//     If these hints are missing, we’ll fail to merge these intervals.
+//     Consequently, once the goal post resets, we’ll find it pointing to an older than expected post.
+//     The workaround is to manually trigger a goal post reset (with the separator controls).
+//     Hints might be missing if we manually browse to a page’s URL.
+//     They might also _go_ missing should Tumblr change their URL patterns.
+//   - When the same context is open in multiple tabs, behaviour may seem unpredictable.
+
+XKit.extensions.separator = (function() {
+	"use strict";
+
+	// Immutables
+	const DASHBOARD_PATH_REGEX = /^\/dashboard(?:\/(?:([2-9]|[1-9]\d+)\/(-?[1-9]\d*)\/?)?)?$/;
```

is there a non-trivial functionality we gain by using [2-9]|[1-9]\d+? I understand that it's more correct, but \d+ seems like it would work just fine in the practical case.

also when do we ever see negative post numbers?

---

¯\\\_(ツ)\_/¯

I figured it was a common enough construct for matching against positive integers, and have a preference towards failing early. And the pattern is unlikely to change with frequency for maintainability to be an issue. (I mean, I feel the escapes and nested optional groups are what make this regex a pain to read.) But I can change it to `\d+`.

Negative post numbers are for backwards paging.